Add mixin to layout constrain with conditional padding

sbrack8t opened the pull request.

sbrack8t committed:
Add mixin to layout constrain with conditional padding

dcmouyard commented on sass/partials/layout/_constrain.scss:
```diff
@@ -1,15 +1,25 @@
 // Styles for constraining the page width.
 
-.l-constrain {
+@mixin constrain($width: em($max-page-width), $padding: true) {
```
I like the idea of adding a mixin for the constrain styles with optional side padding. I think this should go in global/helpers/_constrain.scss instead of in this file though.

dcmouyard commented on sass/partials/layout/_constrain.scss:
```diff
@@ -1,15 +1,25 @@
+  @include constrain;
+}
+
+.l-constrain--global {
```
I don’t see the benefit of adding this variant into Gesso core. Is there a use-case I’m missing that wouldn’t be project-specific?

sbrack8t replied:
I've used this for limiting the entire page width for super large screens

dcmouyard replied:
Ah, okay. 

Would it make sense to have this width be a variable? Also, we would probably only want to add the max-width property since everything else would already be in l-constrain.

sbrack8t replied:
Makes sense to me.

sbrack8t replied:
But we still need the margin auto to center it on the screen

dcmouyard replied:
It would already have margin auto from l-constrain.

dcmouyard commented on sass/partials/layout/_constrain.scss:
```diff
@@ -1,15 +1,25 @@
 .l-constrain--no-padding {
-  padding-left: 0;
-  padding-right: 0;
+  @include constrain(em($max-page-width), false);
```
I think this variant should only have the padding-left/right properties since that’s the only thing this variant overrides.

The only reason to add all of the other styles would be to allow people to only add the l-constrain--no-padding class to an element instead of class="l-constrain l-constrain--no-padding".

sbrack8t replied:
Good point.

sbrack8t committed:
- Add layout constrain break class
- Add default variable to constrain file
- remove mixin from l-constrain variant

sbrack8t committed:
Allow padding as mixin argument

dcmouyard committed:
Updated constrain styles to match coding standards.

dcmouyard commented:
I moved the classes back into the layout partial, but kept the mixins in global so they could be used anywhere else if needed. 

I turned the l-constrain--break variant into a sub-component (l-constrain__break) since it would always be inside an l-constrain parent. This also allow us to easily override the styles based on the l-constrain variant.
